src/components/Settings/test-utils.tsx

```diff
@@ -0,0 +1,67 @@
+import { useContext } from "react";
+import {
+  createSettingsContext,
+  getUseHasUnsavedTab,
+  getUsePopulateForm,
+  SettingsState,
+  useSettingsState,
+} from "./Context";
+
+export type TestRoutes = "foo" | "bar";
+
+export type FormStateMap = {
+  foo: {
+    capsLockEnabled: boolean;
+  };
+  bar: {
+    name: string;
+    age: number;
+  };
+};
+
+export const initialData = {
+  foo: { capsLockEnabled: true },
+  bar: { name: "Sophie", age: 26 },
+};
+
+const formToGqlMap = {
+  foo: (form: FormStateMap["foo"]) => form,
+  bar: (form: FormStateMap["bar"]) => form,
+};
+
+const TestContext = createSettingsContext<TestRoutes, FormStateMap>();
+
+const TestProvider = ({ children }) => {
+  const { getTab, saveTab, setInitialData, tabs, updateForm } =
+    useSettingsState(["foo", "bar"], formToGqlMap);
+
+  return (
+    <TestContext.Provider
```

Comment:
I know this is being used in a test but could we maybe use the workaround here instead of ignoring the rule? https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-no-constructed-context-values.md